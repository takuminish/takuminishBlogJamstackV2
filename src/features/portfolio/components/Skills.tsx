import Typography from "@mui/material/Typography";
import { Skill } from "../../../pages";
import React from "react";
import Box from "@mui/material/Box";
import Grid2 from "@mui/material/Unstable_Grid2";
import Paper from "@mui/material/Paper";
import SkillCard from "./SkillCard";

type Props = {
  skills: Skill[];
};

const CATEGORY = {
  language: "言語",
  framework: "フレームワーク",
};

function Skills(props: Props) {
  const { skills } = props;

  const frameworkSkills = skills.filter(
    (skill) => skill.category[0] === CATEGORY.framework
  );
  const languageSkills = skills.filter(
    (skill) => skill.category[0] === CATEGORY.language
  );

  return (
    <Box>
      <Typography variant="h5" sx={{ fontWeight: 700, textAlign: "center" }}>
        Framework
      </Typography>
      <Grid2 container spacing={3}>
        {frameworkSkills.map((skill) => (
          <Grid2 key={skill.name} xs={12} sm={6} md={4}>
            <SkillCard skill={skill} />
          </Grid2>
        ))}
      </Grid2>

      <Typography variant="h5" sx={{ fontWeight: 700, textAlign: "center" }}>
        Language
      </Typography>
      <Grid2 container spacing={3}>
        {languageSkills.map((skill) => (
          <Grid2 key={skill.name} xs={12} sm={6} md={4}>
            <SkillCard skill={skill} />
          </Grid2>
        ))}
      </Grid2>
    </Box>
  );
}

export default Skills;
