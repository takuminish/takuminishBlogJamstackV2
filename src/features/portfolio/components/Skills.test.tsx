import React from "react";
import { render } from "@testing-library/react";
import Skills, { Props } from "./Skills";

const createDOM = () => {
  const props: Props = {
    skills: [
      {
        fieldId: "skill",
        name: "skill1",
        detailedInfomation: "<p>test</p>",
        yearOfExperience: 3,
        category: ["フレームワーク"],
      },
      {
        fieldId: "skill",
        name: "skill2",
        detailedInfomation: "<p>test</p>",
        yearOfExperience: 3,
        category: ["言語"],
      },
    ],
  };

  return render(<Skills skills={props.skills} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
