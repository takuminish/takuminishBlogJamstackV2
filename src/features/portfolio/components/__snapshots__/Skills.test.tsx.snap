// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-wetkhg-MuiTypography-root"
    >
      Framework
    </h5>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-dkd71f-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-sm-6 MuiGrid2-grid-md-4 css-1edanj2-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
           
          <h6
            class="MuiTypography-root MuiTypography-h6 css-10v34n7-MuiTypography-root"
          >
            skill1
          </h6>
          <div>
            <p>
              test
            </p>
          </div>
        </div>
      </div>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-wetkhg-MuiTypography-root"
    >
      Language
    </h5>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-dkd71f-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-sm-6 MuiGrid2-grid-md-4 css-1edanj2-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
           
          <h6
            class="MuiTypography-root MuiTypography-h6 css-10v34n7-MuiTypography-root"
          >
            skill2
          </h6>
          <div>
            <p>
              test
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
